language/constants_spec.rb

```diff
@@ -425,6 +425,26 @@ module ConstantSpecs
   end
 end
 
+describe "top-level constant lookup" do
+  ruby_version_is "" ... "2.5" do
+    it "searches Object successfully when starts from class" do
+      ->() {
+        String::Hash.should == Hash
+      }.should complain(/toplevel constant Hash referenced by/)
+    end
+  end
+
+  it "searches Object unsuccessfully when starts from module" do
+    ->() { Enumerable::Hash }.should raise_error(NameError)
+  end
+
+  ruby_version_is "2.5" do
+    it "does not search Object after searching other scopes" do
+      ->() { String::Hash }.should raise_error(NameError)
+    end
```

Comment:
Could you move this one below the first spec so the version guards are next to each other?

Reply:
@eregon Done

Reply:
A "when starts from class" is missing here. One nice way would be to move these 2 specs in a nested "on a class" describe block.

Reply:
@eregon done